dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -174,18 +181,34 @@ public DialogueChannel build() {
             // Reloading currently forgets channel state (pinned target, channel scores, concurrency limits, etc...)
             // In a future change we should attempt to retain this state for channels that are retained between
             // updates.
-            LimitedChannel nodeSelectionChannel = new SupplierChannel(cf.uris().map(targetUris -> {
-                reloadMeter.mark();
-                log.info(
-                        "Reloaded channel '{}' targets. (uris: {}, numUris: {}, targets: {}, numTargets: {})",
-                        SafeArg.of("channel", cf.channelName()),
-                        UnsafeArg.of("uris", cf.clientConf().uris()),
-                        SafeArg.of("numUris", cf.clientConf().uris().size()),
-                        UnsafeArg.of("targets", targetUris),
-                        SafeArg.of("numTargets", targetUris.size()));
-                ImmutableList<LimitedChannel> targetChannels = createHostChannels(cf, targetUris);
-                return NodeSelectionStrategyChannel.create(cf, targetChannels);
-            }));
+            LimitedChannel nodeSelectionChannel =
+                    new SupplierChannel(cf.uris().map(new Function<List<TargetUri>, LimitedChannel>() {
+                        private final Map<TargetUri, ChannelState> state = new HashMap<>();
```

Comment:
Let's make this a ConcurrentHashMap for safety, that way we don't need to think about it